### DISCUSSION TITLE
visitorId max length?

### DISCUSSION THREAD

**Discussion by npapagna** [fingerprintjs/fingerprintjs#916] [Q&A]
Do generated `visitorId`s have a maximum length (for storing them in, say, a DB)?

**Comment by makma** [comment]
Are you referring to the Fingerprint Pro `visitorId` or to the `fingerprint` hash generated by Fingerprint OSS?

Fingerprint Pro: The visitorId contains 20 alfanumeric characters (`/^[a-zA-Z0-9]{20}$/.test(visitorId);`)

OSS FingerprintJS: The library generates hashes using murmur hash (murmurX64Hash128). Therefore the output hash value consists of 128 binary digits or 32 hexadecimal characters. Each hexadecimal character represents 4 binary digits, so the output would be a 32-character string. src: https://github.com/fingerprintjs/fingerprintjs/blob/master/src/index.ts#L27

**Reply by Finesse** [reply]
We don't guarantee the max `visitorId` length in FingerprintJS open-source. I don't think we will ever need more than 128 bits of entropy. So I'd say that limiting the `visitorId` field length to 50 characters is a reliable solution.

**Reply by ronaldris21** [reply]
it makes sense, thx!
